src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -150,109 +128,156 @@ private void addListingToBusinesses(PlacesSearchResult place, String url) {
           placeLocation, rating, photos, types, url));
   }
 
-  public void checkIfBusinessesAreBig() {
-    GeoApiContext context = 
-      new GeoApiContext.Builder().apiKey(KEY).build();
-    numberOfSmallBusinesses = 0;
+  public List<Listing> removeBigBusinessesFromResults() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Iterator<Listing> businesses =  allBusinesses.iterator();
+    String businessName = "";
+    int numberOfSmallBusinesses = 0;
 
-    while(businesses.hasNext()
-          && numberOfSmallBusinesses < SMALL_BUSINESSES_DISPLAYED) {
+    while (businesses.hasNext() && 
+            (numberOfSmallBusinesses < SMALL_BUSINESSES_DISPLAYED)) {
       Listing currentBusiness = businesses.next();
-      TextSearchRequest request = 
+
+      if (determineIfTheCurrentBusinessIsBig(currentBusiness)) {
+        businesses.remove();
+      } else {
+          numberOfSmallBusinesses++;
+      }  
+    }
+    return allBusinesses;
+  }
+
+  private boolean determineIfTheCurrentBusinessIsBig(Listing currentBusiness) {
+    if (checkIfBusinessInDatabase(currentBusiness, SMALL_BUSINESSES_DATABASE)) {
+      return false;
+    } else if (checkIfBusinessInDatabase(currentBusiness, BIG_BUSINESSES_DATABASE)) {
+      return true;
+    }
+
+    if (checkNumberOfSimilarBusinesses(currentBusiness) || 
+          checkBusinessThroughLinkedin(currentBusiness)) {
+      addBusinessToDatabase(currentBusiness, BIG_BUSINESSES_DATABASE);
+      return true;
+    }
+
+    addBusinessToDatabase(currentBusiness, SMALL_BUSINESSES_DATABASE);
+    return false;
+  }
+
+  private boolean checkIfBusinessInDatabase(Listing currentBusiness, 
+                                              String databaseEntry) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();    
+    
+    try {
+        String businessName = 
+            (String) datastore.get(KeyFactory.createKey(
+                                    databaseEntry, 
+                                    currentBusiness.getName()))
+                              .getProperty("Business");
+        return true;
+    } catch (EntityNotFoundException e) {
+        // Sometimes the small or big business will be in the database,
+        // but if not additional checks will follow to determine it's size.
+        LOGGER.warning(e.getMessage());
+        return false;
```

Comment:
For consistency, move this outside of the catch block to the end of the function. It won't change any behaviour.

Reply:
Thank you for this suggestion, this makes sense.